aptos-move/aptos-validator-interface/src/lib.rs

```diff
@@ -95,37 +106,71 @@ pub trait AptosValidatorInterface: Sync {
             let addr = *validator_info.account_address();
             result.push((
                 addr,
-                self.get_account_state_by_version(addr, version)?
+                self.get_account_state_by_version(addr, version)
+                    .await?
                     .ok_or_else(|| anyhow!("validator {:?} doesn't exist", addr))?,
             ));
         }
         Ok(result)
     }
 }
 
-pub struct DebuggerStateView<'a> {
-    db: &'a dyn AptosValidatorInterface,
+pub struct DebuggerStateView {
+    query_sender: Mutex<UnboundedSender<(StateKey, Version)>>,
+    query_receiver: Mutex<Receiver<Result<Option<StateValue>>>>,
     version: Option<Version>,
 }
 
-impl<'a> DebuggerStateView<'a> {
-    pub fn new(db: &'a dyn AptosValidatorInterface, version: Option<Version>) -> Self {
-        Self { db, version }
+async fn handler_thread<'a>(
```

Comment:
I had to do a little bit of hack here because the Move's storage api is not `async` and thus we need a new thread to relay the query between async runtime and MoveVM.